### PR TITLE
Improve delete workout DG

### PR DESCRIPTION
**Pull request by Musfirahe0556596**
This PR includes improvements to the delete workout writing and also restructuring of the implementation's table of contents.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -86,8 +86,17 @@ You are now ready to begin developing!
...
+* [Workout](#workout)
+  * [Create New Workout](#create-new-workout)
     * [Design Considerations](#design-considerations-for-creating-a-new-workout) 
+  * [List Workout](#list-workout)
+  * [Delete Existing Workout](#delete-existing-workout)
+    * [Design Considerations](#design-considerations-for-deleting-existing-workout)
+* [Plan](#plan)
+  * [Create A New Plan](#create-a-new-plan)
+  * [List Plans](#list-plans)
+* [Schedule](#schedule)
```
Comment (alanlowzies) [inline docs/DeveloperGuide.md:100]: Great job improving the ToC!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -141,7 +150,11 @@ for subsequent prompts.
 
 ---
 
-### Create New Workout
+### Exercise
+
+---
+### Workout
```
Comment (alanlowzies) [inline docs/DeveloperGuide.md:157]: I think to be safe, can consider adding an extra line between these two lines to avoid any potential rendering issues.

**Review by alanlowzies — APPROVED** [review]
Looks good! Discussed some issues with you on Telegram already :)

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Musfirahe0556596) [reply docs/DeveloperGuide.md:100]: thank you ^_^

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -308,9 +294,99 @@ If the answer given by the user is neither 'yes' nor 'no', user will be prompt t
...
+----------------------------------------------------------------------
+```
+(Steps 12 to 13) The `FileManager#writeNewWorkoutToFile(workoutList)` is called to rewrite
+the `workouts.txt` file according to the newly modified application's workout list.
+<br><br>
+This completes the process of deleting an existing workout in WerkIt!
+
+##### Design Considerations for Deleting Existing Workout
+###### Rewrite All Workout To File
+Currently, the WerkIt! program will rewrite all workout to the resource file, `resource.txt`, when the delete workout
```
[the remark this answers is not in the document]
Comment (Musfirahe0556596) [reply docs/DeveloperGuide.md:367]: yep thanks for noticing this ^_^!

**Pull request merged**